src/BugsnagUnity/Native/Android/NativeInterface.cs

```diff
@@ -320,26 +331,45 @@ public List<Breadcrumb> GetBreadcrumbs()
         AndroidJNI.AttachCurrentThread();
       }
 
-
-      var javaBreadcrumbs = CallNativeObjectMethod("getBreadcrumbs", "()Ljava/util/List;", new object[]{});
-      var iterator = AndroidJNI.CallObjectMethod(javaBreadcrumbs, CollectionIterator, new jvalue[]{});
+      IntPtr javaBreadcrumbs = CallNativeObjectMethodRef("getBreadcrumbs", "()Ljava/util/List;", new object[]{});
+      IntPtr iterator = AndroidJNI.CallObjectMethod(javaBreadcrumbs, CollectionIterator, new jvalue[]{});
+      AndroidJNI.DeleteLocalRef(javaBreadcrumbs);
 
       while (AndroidJNI.CallBooleanMethod(iterator, IteratorHasNext, new jvalue[]{}))
       {
-        var crumb = AndroidJNI.CallObjectMethod(iterator, IteratorNext, new jvalue[]{});
+        IntPtr crumb = AndroidJNI.CallObjectMethod(iterator, IteratorNext, new jvalue[]{});
         breadcrumbs.Add(ConvertToBreadcrumb(crumb));
         AndroidJNI.DeleteLocalRef(crumb);
       }
-
-      AndroidJNI.DeleteLocalRef(javaBreadcrumbs);
```

Comment:
This local ref can still be deleted unless I'm mistaken?

Reply:
Yeah - I just moved the delete to being before the loop as we dont need a collection reference when using an iterator.

Reply:
ah, got it - that's fine.